partials/AmqpConfig.java

```diff
@@ -54,64 +49,49 @@ public ConnectionFactory connectionFactory() {
     }
 
     @Bean
-    public AmqpAdmin amqpAdmin() {
-        return new RabbitAdmin(connectionFactory());
-    }
-
-    @Bean
-    public Declarables exchanges() {
-        return new Declarables(
-                {% for channelName, channel in asyncapi.channels() %}{% if channel.hasSubscribe() %}
-                new TopicExchange({{channelName}}Exchange, true, false){% if not loop.last %},{% endif %}
-                {% endif %}{% endfor %}
-                );
-    }
-
-    @Bean
-    public Declarables queues() {
-        return new Declarables(
-                {% for channelName, channel in asyncapi.channels() %}{% if channel.hasPublish() %}
-                new Queue({{channelName}}Queue, true, false, false){% if not loop.last %},{% endif %}
-                {% endif %}{% endfor %}
-                );
-    }
+    public Declarables declarables() {
+        {% for channelName, channel in asyncapi.channels() %}
+        Queue {{channelName}}_Queue = new Queue({{channelName}}Queue);
+        {% endfor %}
 
-    // consumer
+        {% for channelName, channel in asyncapi.channels() %}
+        TopicExchange {{channelName}}_Exchange = new TopicExchange({{channelName}}Exchange);
+        {% endfor %}
 
-    @Autowired
-    MessageHandlerService messageHandlerService;
-    {% for channelName, channel in asyncapi.channels() %}{% if channel.hasPublish() %}
+        {% for channelName, channel in asyncapi.channels() %}
+        Binding {{channelName}}_Binding = BindingBuilder.bind({{channelName}}_Queue)
+                .to({{channelName}}_Exchange).with({{channelName}}RoutingKey);
+        {% endfor %}
 
-    @Bean
-    public IntegrationFlow {{channelName | camelCase}}Flow() {
-        return IntegrationFlows.from(Amqp.inboundGateway(connectionFactory(), {{channelName}}Queue))
-                .handle(messageHandlerService::handle{{channelName | upperFirst}})
-                .get();
+        return new Declarables(
+                {% set i = 1 %}{% for channelName, channel in asyncapi.channels() %}{% if i == asyncapi.channels() | size %}
+                    {{channelName}}_Queue,
+                    {{channelName}}_Exchange,
+                    {{channelName}}_Binding
+                {% else %}
+                    {{channelName}}_Queue,
```

Comment:
Also here, you could use `{% if not loop.last %},{% endif %}` to add coma after _Binding instead of counting `i`